lib/matplotlib/tight_layout.py

```diff
@@ -173,12 +167,42 @@ def _auto_adjust_subplotpars(
 
 
 @_api.deprecated("3.5")
-@docstring.copy(_auto_adjust_subplotpars)
 def auto_adjust_subplotpars(
         fig, renderer, nrows_ncols, num1num2_list, subplot_list,
         ax_bbox_list=None, pad=1.08, h_pad=None, w_pad=None, rect=None):
-    num1num2_list = [
-        (n1, n1 if n2 is None else n2) for n1, n2 in num1num2_list]
+    """
+    Return a dict of subplot parameters to adjust spacing between subplots
+    or ``None`` if resulting axes would have zero height or width.
+
+    Note that this function ignores geometry information of subplot
+    itself, but uses what is given by the *nrows_ncols* and *num1num2_list*
+    parameters.  Also, the results could be incorrect if some subplots have
+    ``adjustable=datalim``.
+
+    Parameters
+    ----------
+    nrows_ncols : tuple[int, int]
+        Number of rows and number of columns of the grid.
+    num1num2_list : list[int]
+        List of numbers specifying the area occupied by the subplot
```

Comment:
This is actually `list[tuple[int, int]]`, I think?

Reply:
yes